x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/privilege/ClusterPrivilegeResolver.java

```diff
@@ -90,7 +90,9 @@
         "cluster:admin/data_frame/*",
         "cluster:monitor/data_frame/*",
         "cluster:monitor/transform/*",
-        "cluster:admin/transform/*"
+        "cluster:admin/transform/*",
+        "cluster:internal/transform/*",
+        "indices:internal/transform/*"
```

Comment:
`indices:internal/transform/checkpoint` is indexAction in that the request needs index name expansion. This means it cannot be part of a cluster privilege. It really needs be granted as an index privilege. I noticed you already add it as part of `view_index_metadata` index privilege. So I suggest we remove it from here.

For `cluster:internal/transform/*`, I don't know how transform works, so maybe this is normal: But having it here means the end-user needs to have both `manage_transform` cluster privilege and `view_index_metadata` index privilege for the actions to work. Is this expected?

Reply:
> the end-user needs to have both manage_transform cluster privilege and view_index_metadata index privilege for the actions to work. Is this expected?

Yes, that's expected. The 2 actions are a replacement for the already existing checkpointing. It currently uses the index stats API's which require `view_index_metadata`. Note: the old implementation must be kept in order to provide backwards compatibility(e.g. for CCS).

Having that said, I would not mind if we find a solution that does not require `view_index_metadata`. At least in the long run.